lib/cc/cli/help.rb

```diff
@@ -3,30 +3,45 @@
 module CC
   module CLI
     class Help < Command
+      ARGUMENT_LIST = "[commad]".freeze
+      SHORT_HELP = "Show help.".freeze
+      HELP = "#{SHORT_HELP}\n" \
+        "\n" \
+        "    no arguments   Show help summary for all commands.\n" \
+        "    [command]      Show help for specific commands. Can be scpecified multiple times.".freeze
+
       def run
-        say "Usage: codeclimate COMMAND ...\n\nAvailable commands:\n"
-        commands.each do |command|
-          say "    #{command}"
+        if @args.any?
+          @args.each do |command|
+            show_help(command)
+          end
+        else
+          show_help_summary
         end
       end
 
       private
 
-      def commands
-        [
-          "analyze [-f format] [-e engine(:channel)] [path]",
-          "console",
-          "engines:disable #{underline("engine_name")}",
-          "engines:enable #{underline("engine_name")}",
-          "engines:install",
-          "engines:list",
-          "engines:remove #{underline("engine_name")}",
-          "prepare",
-          "help",
-          "init",
-          "validate-config",
-          "version",
-        ].freeze
+      def show_help(command_name)
+        command = Command[command_name]
+        say "Usage: #{$PROGRAM_NAME} #{command.synopsis}\n"
```

Comment:
this `$PROGRAM_NAME` seems to be expanded into a full path:

```
Usage: /usr/src/app/bin/codeclimate analyze [-f format] [-e engine[:channel]] [path]
```

It will always be `codeclimate` so I would be ok with hardcoding that here